docs/ui/Commandk.tsx

```diff
@@ -0,0 +1,106 @@
+import { siteConfig } from '@/lib/config';
+import { Button, Dialog } from '@/ui';
+import { Command, CommandGroup } from 'cmdk';
+import { allContentPages } from 'contentlayer/generated';
+import React from 'react';
+
+import { useRouter } from 'next/navigation';
+
+import { Search } from '@marigold/icons';
+
+export const CommandMenu = () => {
+  // Toggle the menu when ⌘K is pressed
+  const router = useRouter();
+  const [open, setOpen] = React.useState(false);
+  const [search, setSearch] = React.useState('');
+
+  React.useEffect(() => {
+    const handleKeyDown = (e: {
+      key: string;
+      metaKey: boolean;
+      ctrlKey: boolean;
+      preventDefault: () => void;
+    }) => {
+      if (e.key === 'k' && (e.metaKey || e.ctrlKey)) {
+        e.preventDefault();
+        setOpen(open => !open);
+      }
+    };
+    document.addEventListener('keydown', handleKeyDown);
+    return () => document.removeEventListener('keydown', handleKeyDown);
+  }, []);
+
+  const groupedPages = siteConfig.navigation.map(({ name, slug }) => {
+    const items = allContentPages
+      .filter(page => page.slug.includes(slug))
+      .map(({ title, slug, order }) => ({ title, slug, order }));
+    //sort by order if it's defiened, otherwise sort alphabettically
+    items.sort((a, b) => {
+      if (typeof a.order === 'number' && typeof b.order === 'number') {
+        return a.order - b.order;
+      } else {
+        return a.title.localeCompare(b.title);
+      }
+    });
+
+    return { name, slug, items };
+  });
+
+  function redirect(slug: string) {
+    router.push(`/${slug}`);
+    setOpen(false);
+  }
+
+  return (
+    <Dialog.Trigger open={open} onOpenChange={setOpen} dismissable>
+      <Button variant="sunken" size="small" onPress={() => setOpen(true)}>
+        <span className="hidden xl:inline-flex ">Search documentation...</span>
+        <span className="inline-flex xl:hidden ">Search...</span>
+        <div className="hidden h-5 w-10  items-center justify-center rounded-md text-sm lg:inline-flex lg:border lg:border-gray-300 lg:bg-gray-200">
+          <span className="text-xs ">⌘</span>K
```

Comment:
The CMD icon might be weird for non-Mac users. Switch between CMD/Ctrl based on the OS?

Reply:
Also should be a "kbd" not a span :)